server/src/test/java/com/linecorp/centraldogma/server/internal/api/converter/PushMetricTest.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2024 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.centraldogma.server.internal.api.converter;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import com.linecorp.centraldogma.client.CentralDogma;
+import com.linecorp.centraldogma.common.Change;
+import com.linecorp.centraldogma.server.internal.api.ContentServiceV1;
+import com.linecorp.centraldogma.testing.junit.CentralDogmaExtension;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.MeterRegistry;
+
+class PushMetricTest {
+
+    @RegisterExtension
+    static final CentralDogmaExtension dogma = new CentralDogmaExtension();
+
+    @Test
+    void shouldIncreasePushCounter() {
+        final String projectName = "myPro";
+        final String repoName = "myRepo";
+        final MeterRegistry meterRegistry = dogma.dogma().meterRegistry().get();
+        final Counter pushCounter0 =
+                meterRegistry.find("api.requests")
+                             .tags("service", ContentServiceV1.class.getName(), "method", "push")
+                             .tags("project", projectName, "repository", repoName)
+                             .counter();
+        final double before = pushCounter0 != null ? pushCounter0.count() : 0;
+        final CentralDogma client = dogma.client();
+        client.createProject(projectName).join();
+        client.createRepository(projectName, repoName).join();
+        client.forRepo(projectName, repoName)
+              .commit("Add a file", Change.ofTextUpsert("/a.txt", "a"))
+              .push().join();
+
+        await().untilAsserted(() -> {
+            final Counter pushCounter1 =
+                    meterRegistry.find("api.requests")
+                                 .tags("service", ContentServiceV1.class.getName(), "method", "push")
+                                 .tags("project", projectName, "repository", repoName)
+                                 .counter();
```

Comment:
I actually believe only an `AssertionError` is caught.

https://github.com/awaitility/awaitility/blob/fbe16add874b4260dd240108304d5c0be84eabc8/awaitility/src/main/java/org/awaitility/core/AssertionCondition.java#L52-L62

```java
assertThat(pushCounter).isNotNull();
```

Reply:
As other exceptions are rethrown, if `pushCounter` is null, the test will fail.
Anyway, I agree with you to use assertj's detailed error message that would be useful for debugging.

Reply:
> AssertionError instances thrown by the supplier are treated as an assertion failure and proper error message is propagated on timeout. Other exceptions are rethrown immediately as an execution errors.

I now understand your comment after reading the Javadoc of it. TIL.